rum-demo-app-ios/RUM Mobile Demo App/View/ViewControllers/CheckOutVC.swift

```diff
@@ -0,0 +1,282 @@
+
+class CheckOutVC  : UIViewController{
+    
+    @IBOutlet weak var btnPlaceOrder: RoundedCornerButton!
+        
+    @IBOutlet weak var lblCheckOut: UILabel!
+    
+    @IBOutlet weak var txtEmailAddress: DesignableUITextField!
+    
+    @IBOutlet weak var txtStreetAddress: DesignableUITextField!
+    
+    @IBOutlet weak var txtZipCode: DesignableUITextField!
+    
+    @IBOutlet weak var txtCountry: DesignableUITextField!
+    
+    @IBOutlet weak var txtCreditCardNumber: DesignableUITextField!
+    
+    @IBOutlet weak var txtCity: DesignableUITextField!
+    @IBOutlet weak var txtState: DesignableUITextField!
+    
+    @IBOutlet weak var txtCVV: DesignableUITextField!
+    
+    @IBOutlet weak var txtYear: DesignableUITextField!
+    @IBOutlet weak var txtMonth: DesignableUITextField!
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    
+    var checkoutViewModel = CheckOutVM()
+    
+    var dateMonthPicker : UIPickerView!
+    var isMonth : Bool!
+    
+    var arrTextFiled : [txtValidation]! = [.email,.streetAddress,.zipcode,.country,.state,.city,.ccNumber,.month,.year,.cvv]
+    
+    let monthArr = ["01","02","03","04","05","06","07","08","09","10","11","12"]
+    let yearArr = ["2013","2014","2015","2016","2017","2018","2019","2020","2021","2022","2023","2024"]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        dateMonthPicker = UIPickerView()
+        dateMonthPicker.delegate = self
+        dateMonthPicker.dataSource = self
+        txtMonth.inputView = dateMonthPicker
+        txtYear.inputView = dateMonthPicker
+        scrollView.addShadow()
+        
+        txtCreditCardNumber.addTarget(self, action: #selector(didChangeText(textField:)), for: .editingChanged)
+        txtCreditCardNumber.delegate = self
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.addBlueHeader(title: "RUM Mobile Demo", isRightButtonHidden: false, isBackButtonHidden: false)
+        btnPlaceOrder.addTextSpacing()
+        self.addDefaultInputValues()
+        
+        RumEventHelper.shared.trackCustomRumEventFor(.checkout)
+    }
+    
+    @objc func didChangeText(textField:UITextField) {
+        textField.text = self.modifyCreditCardString(creditCardString: textField.text!)
+    }
+    
+    func modifyCreditCardString(creditCardString : String) -> String {
+         let trimmedString = creditCardString.components(separatedBy: "-").joined()
+
+         let arrOfCharacters = Array(trimmedString)
+         var modifiedCreditCardString = ""
+
+         if(arrOfCharacters.count > 0) {
+             for i in 0...arrOfCharacters.count-1 {
+                 modifiedCreditCardString.append(arrOfCharacters[i])
+                 if((i+1) % 4 == 0 && i+1 != arrOfCharacters.count){
+                     modifiedCreditCardString.append("-")
+                 }
+             }
+         }
+         return modifiedCreditCardString
+     }
```

Comment:
This seems to be copypasted from stackoverflow, do we give proper credit in such cases?